cV - 3.0.md

```diff
@@ -0,0 +1,280 @@
+# Correlation Vector v3.0 [Draft]
+
+**Correlation Vector** (a.k.a. **cV**) is a tracing standard that evolved internally at Microsoft, based on a light weight vector clock.
+
+This document covers:
+
+- the additional design goals of the cV that are addressed w.r.t the v2.1 specification
+- the format used on the wire when transmitting the vector across components involved in a trace
+- the protocols used for managing the vector and the specific operators involved
+
+## Design Goals
+
+Here are the key **additional** design goals:
+
+1. Support interop with W3C Distributed Tracing [Link](https://github.com/w3c/distributed-tracing)
+2. Support for vector reset semantics to support traces of arbitrary depth
+3. Support for versioning
+
+## Format
+
+Represented by the following simply grammar ABNF specification
+
+1. vector = ver delim base suffix
+2. ver = "A"
+3. base = 21char lastChar
+4. char = %x30-39 / %x41-5A / %61-7A / %x2F / %2B ; base64 characters
+5. lastChar = "A" / "Q" / "g" / "w" ; base64 characters with four least significant bits of zero
+6. suffix = init / init vector
+7. init = delim tick / sharp id delim tick / dash id delim tick
+8. vector = term / term vector
+9. term = delim tick / under id delim tick
+10. tick = 1*8(%x30-39 / %x41-46) ; hex encoded counter
```

Comment:
Perhaps clarify as base16hex.

If you want to increase the max allowed ticks without increasing the space of encoded data, consider [base32hex](https://en.wikipedia.org/wiki/Base32#base32hex).
I believe the max you could comfortably go with this and maintain lexical sort capability (while also avoid distinguishing upper/lower case) would likely be [base36hex](https://en.wikipedia.org/wiki/Base36).

For example:
4-byte unsigned integer max = 4294967295 (base10) = FFFFFFFF (base16hex) = 3VVVVVV (base32hex) = 1Z141Z3 (base36hex)
base32hex and base36hex each save a character for the same value.

Assuming we stick with 8-chars as our max tick encoding length, base32hex and base36hex provide the following max ticks:
VVVVVVVV (base32hex) = 1,099,511,627,775 (base10)
ZZZZZZZZ (base36hex) = 2,821,109,907,455 (base10)

Reply:
Good suggestion. My only concern is that they are not as popular as hexadecimal. Base36 does not have an RFC and maybe a concern w.r.t. variations introduced during implementation. I've opened a new issue to track this and we can discuss this further.